### ISSUE TITLE
READ BEFORE OPENING AN ISSUE — Common Problems & Solutions

### ISSUE THREAD

**Issue by duskfallcrew** [Ktiseos-Nyx/Ktiseos-Nyx-Trainer#213]
# Before You Open a New Issue...

**Please check this list first!** Most problems people run into are already solved. If your issue is listed here, add a comment to this pinned issue instead of opening a new one.

---

## "Access Denied" / Permission Errors

**This is the #1 issue people hit.** You installed to a restricted location.

**DO NOT install to:**
- `C:\` (drive root)
- `C:\Program Files\` or `C:\Program Files (x86)\`
- `C:\Windows\` or any system directory
- OneDrive, Dropbox, Google Drive, or network drives

**Install to a folder YOU own:**
```
C:\Users\YourName\Projects\Ktiseos-Nyx-Trainer
D:\Projects\Ktiseos-Nyx-Trainer
```

If you already installed to a bad location, **move the entire folder** to a proper location and re-run the installer.

---

## "Module not found" / Import Errors

1. **Did you activate your virtual environment?**
   - Windows: `venv\Scripts\activate`
   - Linux: `source venv/bin/activate`
2. **Did you run the installer?** `python installer.py`
3. **Are you using Python 3.10 or 3.11?** Other versions may have compatibility issues.

---

## Frontend Won't Start / Port 3000 Not Working

1. **Do you have Node.js 18+ installed?** Check with `node --version`
2. **Did npm install succeed?** Check `frontend/node_modules/` exists
3. **On VastAI/RunPod**: Services auto-start. Check logs with `supervisorctl status` or `ps aux | grep node`

---

## CUDA Out of Memory

- **Reduce batch size** (try 1)
- **Enable gradient checkpointing**
- **Lower resolution** (e.g., 512 instead of 1024 for SDXL)
- **Use fp16 or bf16 mixed precision**
- 12GB VRAM is the minimum. 24GB recommended for SDXL.

---

## Training Loss Shows NaN

- **Lower your learning rate** (try 1e-4 or lower)
- **Check your dataset** — corrupt images or empty caption files cause this
- **Try a different optimizer** (AdamW8bit is a safe default)

---

## WD14 Tagging Issues

The tagger uses **Python onnxruntime-gpu** with CUDA acceleration on all platforms (local, VastAI, RunPod). If tagging is slow or failing:

- **Check CUDA is working**: `python -c "import onnxruntime; print(onnxruntime.get_available_providers())"`  — should show `CUDAExecutionProvider`
- **Check the installer ran**: The installer installs `onnxruntime-gpu` from the CUDA 12 index
- **If onnxruntime-gpu failed to install**: You may be missing CUDA 12.1+ or cuDNN. Check with `nvidia-smi`

---

## Gated Models (Flux, SD3.5) Won't Download

These models require:
1. A **HuggingFace account**
2. **Accept the model license** on the model's HuggingFace page
3. **Set your HF token** in Settings (get one from [huggingface.co/settings/tokens](https://huggingface.co/settings/tokens))

---

## The Diagnostic Tool

Before reporting a bug, **run the diagnostic tool** and include the output:

```bash
python diagnose.py
```

This gives us your Python version, CUDA status, installed packages, and common issues — all in one shot.

---

## How to Report a Bug Properly

If your issue ISN'T listed above, open a new issue with:

1. **What you expected** vs **what actually happened**
2. **Full error message** (copy-paste, not a screenshot of text)
3. **Your environment**: OS, GPU, Python version, local or cloud (VastAI/RunPod)
4. **Diagnostic output** from `python diagnose.py`
5. **Steps to reproduce** — what did you click/run?

"It doesn't work" without details = we can't help you. We want to help, but we need the info!

---

## Useful Links

- [README & Installation Guide](https://github.com/Ktiseos-Nyx/Ktiseos-Nyx-Trainer#quick-start)
- [Troubleshooting Guide](https://github.com/Ktiseos-Nyx/Ktiseos-Nyx-Trainer/blob/main/documentation/troubleshooting.md)
- [Discord](https://discord.gg/HhBSM9gBY) — for quick questions and chat
- [Development Status](https://github.com/Ktiseos-Nyx/Ktiseos-Nyx-Trainer/blob/main/STATUS.md)

---

**If your problem IS listed here**, please comment on this issue instead of opening a new one. Thanks for helping keep things tidy!
